### PR TITLE
Issue #4 - Robot->NUC Feedback

### PR DESCRIPTION
**Pull request by matthew-conner**
I believe this will have fixed all compile errors.

### CODE REVIEW

**Comment by anidev** [comment]
(just adding note) partially resolves #4, waiting for API finish in VESC_BBB_UART

File: src/robot2017/src/robot_exec.cpp [hunk 1]
```diff
@@ -57,6 +59,7 @@ void RobotExec::autonomyReceived(const robot_msgs::Autonomy& cmd)
     {
         autonomyExec(cmd);
     }
+    publishMotors();
```
Comment (anidev) [inline src/robot2017/src/robot_exec.cpp:68]: Good for now but see if you can call this via a ROS timer so it does not require receiving input from autonomy

**Review by anidev — CHANGES_REQUESTED** [review]
Compiles fine, however running it fails:

````
[FATAL] [1490717077.957181424]: You must call ros::init() before creating the first NodeHandle
[ERROR] [1490717077.957936877]: [registerPublisher] Failed to contact master at [:0].  Retrying...
````

Also includes error
````
Couldn't find an AF_INET address for []
````
but likely as a result of the above issue.

**Comment by anidev** [comment]
All right so before merging this, could you run a test on the BeagleBone or a VM? Either change the code temporarily so it calls publishMotors somewhere just for testing, or more ideally change it so publishMotors is called via a ROS timer as I mentioned above.

**Comment by matthew-conner** [comment]
I don't have a VM setup but I can try to get it running on the Beaglebone. Didn't we have a setup where we could test everything on one computer though? Assuming we haven't made any major changes, running test.launch should allow us to see the communication, right? If so, it probably would be simplest to just add a subscriber to the test_controller file that already was publishing teleop commands.

**Comment by anidev** [comment]
You can use rostopic to see the data being published, but since right now publishMotors is only called when it receives autonomy data you either need to change that or send autonomy data via the test controller.

If you have access to a Linux computer or VM of any sort, you can use docker for testing - I have a docker compose script that will automatically set up and connect two docker containers for testing ROS.

**Comment by matthew-conner** [comment]
No yeah I have a linux setup obviously, but it's dual boot instead of a VM, so I just meant I couldn't use multiple instances of it. I've never used docker before but I doubt it's too difficult, especially if you already have a script setup.

In regards to the MotorFeedback publisher: I agree that its simplest for testing to bypass the whole "autonomy received" part of the program, potentially using the timer feature. But for the final design - will we still want it to be done seperately, or have it be called after autonomy sends a command, since if autonomy isn't active they won't actually need any feedback.

**Comment by anidev** [comment]
That's a good point though the way it is now, autonomy is always sending us
commands, we just choose to ignore them if not in autonomy mode. So then I
guess it doesn't really matter either way, so for now it's fine the way it
is. I think best solution for testing is have test node send autonomy
commands as well. You could even set it up so it sends a teleop command to
enable autonomy mode so we don't have to hard code autonomy true, just to
test that works too.

On Mar 30, 2017 9:32 AM, "Matthew Conner" <notifications@github.com> wrote:

No yeah I have a linux setup obviously, but it's dual boot instead of a VM,
so I just meant I couldn't use multiple instances of it. I've never used
docker before but I doubt it's too difficult, especially if you already
have a script setup.

In regards to the MotorFeedback publisher: I agree that its simplest for
testing to bypass the whole "autonomy received" part of the program,
potentially using the timer feature. But for the final design - will we
still want it to be done seperately, or have it be called after autonomy
sends a command, since if autonomy isn't active they won't actually need
any feedback.

—
You are receiving this because your review was requested.

Reply to this email directly, view it on GitHub
<https://github.com/VTAstrobotics/Robot-2017/pull/9#issuecomment-290411522>,
or mute the thread

**Comment by matthew-conner** [comment]
Everything should be done at this point.

**Review by anidev — APPROVED** [review]

**Comment by anidev** [comment]
Looks good, though could you clean up the info stream messages first before merging since it prints a lot of messages to console very quickly

**Comment by matthew-conner** [comment]
Decreased ros rate so that ros logging statements (printed to console) occur less frequently. Obviously we will want to increase the rate again when we are actually testing the autonomy routine.

**Pull request merged**